'RealmProxy is not part of the schema' on 10.6.0+ in multi module projects

### How frequently does the bug occur?

All the time

### Description

When using a RealmModule from a library, since Realm version 10.6.0 (up until 10.10.1) doing _copyToRealmOrUpdate()_ with a managed RealmObject crashes the App.

With Realm version v10.5.1 (or earlier) it works. 
Without the RealmModule it works.

We found out that prior 10.6.0 Realm did call _DefaultRealmModuleMediator.hasPrimaryKeyImpl_ instead of _CompositeMediator.hasPrimaryKeyImpl_, despite the file _DefaultRealmModuleMediator_ is also created in 10.6.0 upwards.

### Stacktrace & log output

```shell
Caused by: io.realm.exceptions.RealmException: org_xyz_RealmObjectRealmProxy is not part of the schema for this Realm
        at io.realm.internal.modules.CompositeMediator.getMediator(CompositeMediator.java:205)
        at io.realm.internal.modules.CompositeMediator.hasPrimaryKeyImpl(CompositeMediator.java:111)
        at io.realm.internal.RealmProxyMediator.hasPrimaryKey(RealmProxyMediator.java:111)
        at io.realm.Realm.hasPrimaryKey(Realm.java:1836)
        at io.realm.Realm.checkHasPrimaryKey(Realm.java:1829)
        at io.realm.Realm.copyToRealmOrUpdate(Realm.java:1187)
```


### Can you reproduce the bug?

Yes, always

### Reproduction Steps

I can provide a minimal test project.

Here are the important parts.
Simple RealmObject in app-module:

```
@RealmClass
open class Task : RealmObject() {
    @PrimaryKey
    var id: Long? = null

    var name: String? = null
}
```

Simple RealmObject in lib-module:

```
@RealmClass
open class LibTask: RealmObject() {
    var id: Long? = null
}
```

With RealmModule:

```
@RealmModule(library = true, allClasses = true) // same behavior with classes = [TestTask::class] instead
class RealmModuleFromLib
```

Application class:

```
class ModuleTestApp: Application() {

    override fun onCreate() {
        super.onCreate()
        Realm.init(this)

        val config = RealmConfiguration.Builder()
            .name("db")
            .allowQueriesOnUiThread(true)
            .allowWritesOnUiThread(true)
            .modules(Realm.getDefaultModule(), RealmModuleFromLib())
            .build()

        Realm.setDefaultConfiguration(config)

        RealmLog.setLevel(LogLevel.ALL)
    }
}
```

Insert this code anywhere:

```
Realm.getDefaultInstance().use { readRealm ->

            readRealm.executeTransaction { writeRealm ->
                Log.d("", "Create and insert new Task.")
                writeRealm.copyToRealmOrUpdate(Task().apply { this.id = 123L })
            }

            val task = readRealm.where<Task>().findFirst()
            Log.d("", "Retrieve managed Task (Id/Name): '${task?.id}/${task?.name}'")

            readRealm.executeTransaction { writeRealm ->
                Log.d("", "Edit Task and copyToRealmOrUpdate...")
                task?.name = "Test"
                writeRealm.copyToRealmOrUpdate(task)
            }

            Log.d("", "Show name of edited Task: '${task?.name}'")
        }
```

### Version

10.6.0 upwards

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

Android minSDK 24, targetSDK 31

### Build environment

Android Studio Bumblebee | 2021.1.1 Patch 2
Runtime version: 11.0.11+0-b60-7772763 aarch64, tried it also with 1.8
Android Build Tools version: 31.0.0
Android Gradle Tools: 7.1.2
Gradle version: 7.2

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,7 +4,7 @@
 * Throw a more comprehensive error when initializing Realm on an Instant App.
 
 ### Fixed
-* None.
+* Fix issues resolving class information for `copyToRealmOrUpdate` for already managed objects in multi module projects. (Issue [#7650](https://github.com/realm/realm-java/issues/7650))
 
 ### Compatibility
 * File format: Generates Realms with format v22. Unsynced Realms will be upgraded from Realm Java 2.0 and later. Synced Realms can only be read and upgraded if created with Realm Java v10.0.0-BETA.1.

--- a/realm/realm-library/src/main/java/io/realm/internal/modules/CompositeMediator.java
+++ b/realm/realm-library/src/main/java/io/realm/internal/modules/CompositeMediator.java
@@ -200,7 +200,7 @@ public class CompositeMediator extends RealmProxyMediator {
 
     // Returns the mediator for a given model class (not RealmProxy) or throws exception
     private RealmProxyMediator getMediator(Class<? extends RealmModel> clazz) {
-        RealmProxyMediator mediator = mediators.get(clazz);
+        RealmProxyMediator mediator = mediators.get(Util.getOriginalModelClass(clazz));
         if (mediator == null) {
             throw new RealmException(clazz.getSimpleName() + " is not part of the schema for this Realm");
         }

